### PR TITLE
[WIP] feat: #1904 wip

### LINKED ISSUE #1904: [FEATURE] create new section 'templates' in atoms/molecules/organisms section
**Describe how this feature will help us improve**
Describe why do you think this feature is needed. 

Currently for pages and examples we sometimes use internal components that could be used as templates in atomic design
https://github.com/vuestorefront/storefront-ui/blob/develop/packages/vue/src/examples/pages/checkout/_internal/ConfirmOrder.vue

It would be beneficial to have these components also available as a part of Storefront UI Components Library.

1. Create new folder in `packages/vue/src/components` called `templates`
2. Copy internal example/page components there and change their names to i.e. `SfConfirmOrder`
3. Update docs with a new page `templates`
4. Update examples to use new templates components from @storefront-ui/vue rather than from _internal directory

### PR DESCRIPTION
**Pull request by Baroshem**
# Related issue

Closes #1904 

# Scope of work

# Screenshots of visual changes

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/templates/SfShipping/SfShipping.stories.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { SfShipping } from "@storefront-ui/vue"
+import { countries, shippingMethodsData } from "../internalData.js"
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfShipping/SfShipping.stories.js:2]: There is also no ShippingMethods in internalData file.
Screenshot from 2021-09-28 16-04-47

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      <SfHeading
+        :title="reviewTitle"
+        :level="reviewTitleLevel"
+        class="
+          sf-heading--left sf-heading--no-underline
+          sf-order-review__heading
+        "
+      />
+    </slot>
+    <slot name="personal-details">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:13]: Add v-bind here also.

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      </p>
+      <br />
+      <p class="sf-order-review__content">
+        {{ order.email }}
+      </p>
+      <p class="sf-order-review__content">
+        {{ shipping.phoneNumber }}
+      </p>
+    </slot>
+    <slot name="shipping-details">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:38]: Add v-bind here also.

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+          }}</span
+          ><br />
+          {{ payment.streetName }} {{ payment.apartment }}, {{ payment.zipCode
+          }}<br />
+          {{ payment.city }}, {{ payment.country }}
+        </p>
+        <p class="sf-order-review__content">{{ payment.phoneNumber }}</p>
+      </template>
+    </slot>
+    <slot name="payment-details">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:87]: Add v-bind here also.

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+          class="sf-order-review__promo-code-button"
+          data-testid="apply-button"
+          @click="$emit('click:promo-code-apply')"
+        >
+          Apply
+        </SfButton>
+      </slot>
+    </div>
+    <div class="sf-order-review__characteristics">
+      <slot name="characteristics">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:118]: Add v-bind here also.

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue [hunk 6]
```diff
@@ -0,0 +1,170 @@
+<template>
+  <div class="sf-order-summary">
+    <slot name="heading">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue:3]: Add v-bind here also.

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        :title="orderTitle"
+        :level="orderTitleLevel"
+        class="
+          sf-heading--left sf-heading--no-underline
+          sf-order-summary__heading
+        "
+      />
+    </slot>
+    <div class="highlighted highlighted--total">
+      <slot name="summary">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue:14]: Add v-bind here also.

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+          class="sf-order-summary__promo-code-button"
+          data-testid="apply-button"
+          @click="$emit('click:apply-code')"
+        >
+          Apply
+        </SfButton>
+      </slot>
+    </div>
+    <div class="sf-order-summary__characteristics">
+      <slot name="characteristics">
```
Comment (AdamPawlinski) [inline packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue:68]: Add v-bind here also.

File: packages/vue/src/components/pages/checkout/Checkout.vue [hunk 9]
```diff
@@ -79,25 +79,34 @@
...
+  SfSteps,
+  SfButton,
+  SfPersonalDetails,
+  SfShipping,
+  SfPayment,
+  SfConfirmOrder,
+  SfOrderSummary,
+  SfOrderReview,
+} from "@storefront-ui/vue"
+// import {
```
Comment (justyna-13) [inline packages/vue/src/components/pages/checkout/Checkout.vue:92]: comments

File: packages/vue/src/components/pages/checkout/Checkout.vue [hunk 10]
```diff
@@ -270,7 +279,7 @@ export default {
           icon: "return",
         },
       ],
-    };
```
Comment (justyna-13) [inline packages/vue/src/components/pages/checkout/Checkout.vue:273]: please don't remove semicolons

File: packages/vue/src/components/pages/checkout/Checkout.vue [hunk 11]
```diff
@@ -279,18 +288,18 @@ export default {
         ...this.personalDetails,
         shipping: { ...this.shipping },
         payment: { ...this.payment },
-      };
```
Comment (justyna-13) [inline packages/vue/src/components/pages/checkout/Checkout.vue:282]: please don't remove semicolons

File: packages/vue/src/components/pages/checkout/Checkout.vue [hunk 12]
```diff
@@ -279,18 +288,18 @@ export default {
...
   methods: {
     updateStep(next) {
       // prevent to move next by SfStep header
       if (next < this.currentStep) {
-        this.currentStep = next;
+        this.currentStep = next
       }
     },
   },
-};
```
Comment (justyna-13) [inline packages/vue/src/components/pages/checkout/Checkout.vue:293]: please don't remove semicolons

File: packages/vue/src/components/pages/checkout/Checkout.vue [hunk 13]
```diff
@@ -279,18 +288,18 @@ export default {
...
         payment: { ...this.payment },
-      };
+      }
     },
   },
   methods: {
     updateStep(next) {
       // prevent to move next by SfStep header
       if (next < this.currentStep) {
-        this.currentStep = next;
```
Comment (justyna-13) [inline packages/vue/src/components/pages/checkout/Checkout.vue:289]: please don't remove semicolons

File: packages/vue/src/components/pages/detailed-cart/DetailedCart.vue [hunk 14]
```diff
@@ -98,8 +98,9 @@ import {
   SfProperty,
   SfHeading,
   SfBreadcrumbs,
-} from "@storefront-ui/vue";
-import { OrderSummary } from "./_internal/index.js";
+  SfCartOrderSummary,
+} from "@storefront-ui/vue"
+// import { OrderSummary } from "./_internal/index.js";
```
Comment (justyna-13) [inline packages/vue/src/components/pages/detailed-cart/DetailedCart.vue:103]: comment

File: packages/vue/src/components/pages/detailed-cart/DetailedCart.vue [hunk 15]
```diff
@@ -209,23 +210,23 @@ export default {
...
-    };
+    }
   },
   computed: {
     totalItems() {
       return this.products.reduce(
         (totalItems, product) => totalItems + parseInt(product.qty, 10),
         0
-      );
+      )
```
Comment (justyna-13) [inline packages/vue/src/components/pages/detailed-cart/DetailedCart.vue:220]: please don't remove semicolons in any file

File: packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.md [hunk 16]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+              class="sf-confirm-order__product-price"
+            />
+          </SfTableData>
+        </SfTableRow>
+      </SfTable>
+    </slot>
+    <div class="sf-confirm-order__totals">
+      <slot name="summary">
+        <SfProperty
+          name="Subtotal"
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue:76]: it should also be possible to change this name in props, wdyt?

File: packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+          :value="subtotal"
+          class="
+            sf-property--full-width
+            sf-confirm-order__property sf-confirm-order__property-subtotal
+          "
+        >
+        </SfProperty>
+        <SfProperty
+          name="Shipping"
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue:85]: as above and for next properties too

File: packages/vue/src/components/templates/SfMyNewsletter/SfMyNewsletter.md [hunk 19]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfMyNewsletter/SfMyNewsletter.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfMyProfile/SfMyProfile.md [hunk 20]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfMyProfile/SfMyProfile.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfOrderHistory/SfOrderHistory.md [hunk 21]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderHistory/SfOrderHistory.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.md [hunk 22]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue [hunk 23]
```diff
@@ -0,0 +1,215 @@
+<template>
+  <div class="sf-order-review" data-testid="review">
+    <slot name="heading">
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:3]: v-bind

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue [hunk 24]
```diff
@@ -0,0 +1,215 @@
...
+    </slot>
+    <slot name="shipping-details">
+      <div class="sf-order-review__header">
+        <p class="sf-order-review__title">Shipping details</p>
+        <SfButton
+          class="sf-button--text"
+          data-testid="shipping-edit-button"
+          @click="$emit('click:shipping-details-edit', 1)"
+        >
+          Edit
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:46]: what about changing buttons name by props?

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.md [hunk 25]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.stories.js [hunk 26]
```diff
@@ -0,0 +1,123 @@
...
+const Template = (args, { argTypes }) => ({
+  props: Object.keys(argTypes),
+  components: { SfOrderSummary },
+  template: `
+  <SfOrderSummary
+    :order-title="orderTitle"
+    :order-title-level="orderTitleLevel"
+    :order="order"
+    :characteristics="characteristics"
+  >
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.stories.js:118]: there's nothing inside this component -  <SfOrderSummary :order-title="orderTitle" ... />

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue [hunk 27]
```diff
@@ -0,0 +1,170 @@
...
+        class="
+          sf-heading--left sf-heading--no-underline
+          sf-order-summary__heading
+        "
+      />
+    </slot>
+    <div class="highlighted highlighted--total">
+      <slot name="summary">
+        <SfProperty
+          name="Products"
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue:16]: properties names could be changed by props

File: packages/vue/src/components/templates/SfPayment/SfPayment.md [hunk 28]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfPayment/SfPayment.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfPersonalDetails/SfPersonalDetails.md [hunk 29]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfPersonalDetails/SfPersonalDetails.md:1]: we don't need this file

File: packages/vue/src/components/templates/SfShipping/SfShipping.md [hunk 30]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfShipping/SfShipping.md:1]: we don;t need this file

File: packages/vue/src/components/templates/SfShippingDetails/SfShippingDetails.md [hunk 31]
```diff
@@ -0,0 +1,5 @@
+# component-description
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfShippingDetails/SfShippingDetails.md:1]: we don;t need this file

File: packages/vue/src/components/templates/SfShippingDetails/SfShippingDetails.vue [hunk 32]
```diff
@@ -0,0 +1,349 @@
...
+        data-testid="shipping-details-tabs"
+      >
+        <SfTab :title="changeAddressTabTitle">
+          <slot name="change-address-description">
+            <p class="message">
+              {{ changeAddressDescription }}
+            </p>
+          </slot>
+          <div class="form">
+            <slot name="form">
```
Comment (justyna-13) [inline packages/vue/src/components/templates/SfShippingDetails/SfShippingDetails.vue:18]: v-bind

File: packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue (hunk 17 above)
Comment (Baroshem) [reply packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue:76]: Add to issue with customization of templates

File: packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue (hunk 27 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/templates/SfOrderSummary/SfOrderSummary.vue:16]: Thanks good point :)

File: packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue (hunk 24 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/templates/SfOrderReview/SfOrderReview.vue:46]: thanks :)

File: packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue (hunk 18 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/templates/SfConfirmOrder/SfConfirmOrder.vue:85]: Thanks :)

**Review by AdamPawlinski — APPROVED** [review]

**Pull request merged**